smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/waiters/WaiterAcceptorGenerator.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.swift.codegen.waiters
+
+import software.amazon.smithy.jmespath.JmespathExpression
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.core.CodegenContext
+import software.amazon.smithy.swift.codegen.model.toUpperCamelCase
+import software.amazon.smithy.swift.codegen.utils.toLowerCamelCase
+import software.amazon.smithy.waiters.Acceptor
+import software.amazon.smithy.waiters.Matcher
+import software.amazon.smithy.waiters.PathComparator
+import software.amazon.smithy.waiters.PathMatcher
+
+class WaiterAcceptorGenerator(
+    val writer: SwiftWriter,
+    val ctx: CodegenContext,
+    val service: ServiceShape,
+    val waitedOperation: OperationShape,
+    val acceptor: Acceptor,
+) {
+
+    fun render() {
+        val inputType = waitedOperation.inputShape.name
+        val outputType = waitedOperation.outputShape.name
+        writer.openBlock(".init(state: .${acceptor.state}) { (input: $inputType, result: Result<$outputType, Error>) -> Bool in", "},") {
+            val matcher = acceptor.matcher
+            when (matcher) {
+                is Matcher.SuccessMember -> {
+                    writer.openBlock("switch result {", "}") {
+                        writer.write("case .success: return ${"true".takeIf { matcher.value } ?: "false"}")
+                        writer.write("case .failure: return ${"false".takeIf { matcher.value } ?: "true"}")
+                    }
+                }
+                is Matcher.OutputMember -> {
+                    renderInputOutputBlockContents(false, matcher.value)
+                }
+                is Matcher.InputOutputMember -> {
+                    renderInputOutputBlockContents(true, matcher.value)
+                }
+                is Matcher.ErrorTypeMember -> {
+                    val errorTypeName = "${waitedOperation.toUpperCamelCase()}OutputError"
+                    var errorEnumCaseName = "${matcher.value.toLowerCamelCase()}"
+                    writer.write("// Match on error case: $errorEnumCaseName")
+                    writer.write("return false")
```

Comment:
The `errorType` matcher is essentially disabled (i.e. it never matches) while I determine the best way to match errors; the approach commented out below is too strict.

`errorType` matching will be implemented in a future PR.

Reply:
`errorType` is now implemented.